third_party/txt/src/minikin/LineBreaker.cpp

```diff
@@ -361,7 +361,8 @@ void LineBreaker::pushBreak(int offset, float width, uint8_t hyphenEdit) {
 // with nullptr as the paint property, which will lead it to assume the width
 // has already been calculated. Used for properly breaking inline widgets.
 void LineBreaker::setCustomCharWidth(size_t offset, float width) {
-  mCharWidths[offset] = (width);
+  mCharWidths.reserve(offset + 1);
+  mCharWidths[offset] = width;
```

Comment:
The size of `mCharWidths` should match the size of the text buffer.

When would `setCustomCharWidth` receive an offset greater than the text size?  If that happens, then should libtxt have called `LineBreaker::resize` first?

Reply:
Ahh maybe the better fix is to ensure LineBreaker::resize is being called appropriately.